### PR TITLE
feat(Button): New `variants` and minor fixes

### LINKED ISSUE #821: Button: New variants
Variant prop should include: 
- `Filled`, `outline`, `quiet` should be replaced by `primary`, `secondary` and `tertiary`. 

Color prop should include: 
- `Primary` and `secondary` should be replaced by `default` and `neutral` (in addition to `success` and `danger` that already exist)

### PR DESCRIPTION
**Pull request by mimarz**
resolves #821 

- Renamed variants from `'filled' | 'outline' | 'quiet'` to `'primary' | 'secondary' | 'tertiary'`
- Updated hardcoded color values to use correct tokens (https://github.com/digdir/designsystem/issues/604)
- Removed smaller left/right padding on icon button to make the `Button` properly square

### CODE REVIEW

File: packages/react/src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -70,88 +49,194 @@ export const Preview: Story = {
...
+      variant='primary'
+      aria-label='Primary med ikon'
     ></Button>
     <Button
       icon={icon}
-      variant='outline'
-      aria-label=' Omriss med ikon'
+      color='second'
+      variant='secondary'
+      aria-label=' Sekundar med ikon'
```
Comment (Barsnes) [inline packages/react/src/components/Button/Button.stories.tsx:161]: Typo, should be `Sekundær` or `Secondary`

File: packages/react/src/components/Button/Button.stories.tsx [hunk 2]
```diff
@@ -70,88 +49,194 @@ export const Preview: Story = {
...
+      icon={<akselIcons.CheckmarkIcon />}
+      color='success'
+      variant='primary'
+      aria-label='Primary med ikon'
+    ></Button>
+    <Button
+      icon={<akselIcons.CheckmarkIcon />}
+      color='success'
+      variant='secondary'
+      aria-label=' Sekundar med ikon'
```
Comment (Barsnes) [inline packages/react/src/components/Button/Button.stories.tsx:179]: Typo, should be `Sekundær` or `Secondary`

File: packages/react/src/components/Button/Button.stories.tsx [hunk 3]
```diff
@@ -70,88 +49,194 @@ export const Preview: Story = {
...
+      icon={<akselIcons.TrashIcon />}
+      color='danger'
+      variant='primary'
+      aria-label='Primary med ikon'
+    ></Button>
+    <Button
+      icon={<akselIcons.TrashIcon />}
+      color='danger'
+      variant='secondary'
+      aria-label=' Sekundar med ikon'
```
Comment (Barsnes) [inline packages/react/src/components/Button/Button.stories.tsx:197]: Typo, should be `Sekundær` or `Secondary`

File: packages/react/src/components/Button/Button.stories.tsx [hunk 4]
```diff
@@ -70,88 +49,194 @@ export const Preview: Story = {
...
       icon={icon}
-      variant='filled'
-      aria-label='Fylt med ikon'
+      variant='primary'
+      aria-label='Primary med ikon'
+    ></Button>
+    <Button
+      icon={icon}
+      variant='secondary'
+      aria-label=' Sekundar med ikon'
```
Comment (Barsnes) [inline packages/react/src/components/Button/Button.stories.tsx:144]: Typo, should be `Sekundær` or `Secondary`

File: packages/react/src/components/Button/Button.stories.tsx [hunk 5]
```diff
@@ -70,88 +49,194 @@ export const Preview: Story = {
...
-};
+export const UsedAsLink: StoryFn<typeof Button> = () => (
+  <Button
+    as='a'
+    icon={<akselIcons.PersonChatIcon />}
+    iconPlacement='right'
+    target='_blank'
+    href='https://www.designsystemet.no'
+  >
+    Bli med på Slack
```
Comment (Barsnes) [inline packages/react/src/components/Button/Button.stories.tsx:65]: Maybe change the text here to what the link actually is?
Not a big issue, but would be nice.

Comment (Barsnes) [reply packages/react/src/components/Button/Button.stories.tsx:65]: Change the text to `Gå til Designsystemet.no`?

**Review by Barsnes — CHANGES_REQUESTED** [review]
If the `aria-label` on `secondary` is norwegian, so should `primary` and `tertiary` as well.
Primary = Primær
Tertiary = Tertiær

**Comment by mimarz** [comment]
> If the `aria-label` on `secondary` is norwegian, so should `primary` and `tertiary` as well. Primary = Primær Tertiary = Tertiær

Good point, but did it the other way around as to have the name match the variant.
Renamed `Sekundær` to `Secondary` on the labels

**Review by Barsnes — APPROVED** [review]
👌

**Pull request merged**